src/prefect/server/schemas/graph.py

```diff
@@ -19,6 +19,7 @@ class GraphArtifact(PrefectBaseModel):
     key: Optional[str]
     type: str
     is_latest: bool
+    data: Optional[float]  # we only return data for progress artifacts for now
```

Comment:
Are there other artifacts for which we might return `data` in the future?

If we type `data` as `Optional[float]`, any non-float value returned in the future is technically a breaking API change

Reply:
Yes that is a possibility -- we could type this as `Optional[Any]`